src/stream/futures_unordered.rs

```diff
@@ -0,0 +1,103 @@
+use future::{Future, IntoFuture};
+use stream::Stream;
+use poll::Poll;
+use Async;
+use stack::{Stack, Drain};
+use std::sync::Arc;
+use task::{self, UnparkEvent};
+
+use std::prelude::v1::*;
+
+/// An adaptor for a stream of futures to execute the futures concurrently, if
+/// possible, delivering results as they become available.
+///
+/// This adaptor will return their results in the order that they complete.
+/// This is created by the `futures` method.
+///
+#[must_use = "streams do nothing unless polled"]
+pub struct FuturesUnordered<F>
+    where F: Future
+{
+    futures: Vec<Option<F>>,
+    stack: Arc<Stack<usize>>,
+    pending: [Option<Drain<usize>>; 2],
```

Comment:
I think there only needs to be one of these, right? Either both of these will be `None` or one will be.